### PR TITLE
An attempt to reduce singulo lag from power wires destruction.

### PR DESCRIPTION
**Pull request by phil235**
When the singularity destroys X wires at once, the game used to rebuild the power network X times, after each individual wire is destroyed. Now by adding a spawn(0) and checking whether the adjacent connected wires are still there before calling propagate_network() that proc is only called after the destruction and only by the surviving wires that were connected to the ones destroyed.

Also fixes runtimes when atmos pipe stuff is destroyed.

### CODE REVIEW

**Comment by Xhuis** [comment]
つ ◕_◕ ༽つ PHIL OUR LORD AND SAVIOR つ ◕_◕ ༽つ

**Comment by PKPenguin321** [comment]
ty based phil

**Comment by RocKeD** [comment]
THANK YOU PHIL, holy crap he works fast.

File: code/ATMOSPHERICS/pipes.dm [hunk 1]
```diff
@@ -27,7 +27,10 @@
 		air_update_turf()
 
 /obj/machinery/atmospherics/pipe/return_air()
-	return parent.air
+	if(parent)
+		return parent.air
+	else
+		return null
```
Comment (Aranclanos) [inline code/ATMOSPHERICS/pipes.dm]: pipes are set to always have a parent, don't do this

Comment (RemieRichards) [inline code/ATMOSPHERICS/pipes.dm]: On top of that, most of Atmos' runtimes are deliberately tied to scenarios we don't want, meaning the runtime is our sign that something <i>else</i> is broken.

**Comment by Aranclanos** [comment]
work on your commit logs

**Pull request closed without merging**